Sources/CasePaths/Operators.swift

```diff
@@ -26,6 +26,12 @@ public prefix func / <Root, Value>(
   .case(embed)
 }
 
+public prefix func / <Root, Value>(
+  embed: @escaping (Value) -> Root?
+) -> CasePath<Root?, Value> {
+  .optionalCase(embed)
```

Comment:
(Would have to rename this:)
```suggestion
  .case(embed)
```

Reply:
If it works for you, ok. I seem to recall when I tried overloading the name, it called the non-optional version somewhere that it should have called the optional version.

Reply:
Just wondering how that would be possible since the module-level `extract` function and `/` operators seem to overload just fine. Is there something strange with the generic context? Can you cook up a failure that I can take a look at?

Reply:
I just double-checked. If you rename `optionalCase` to `case`, the `/` overload that currently calls `optionalCase` calls the wrong overload of `case`.

Reply:
... and then it hits the “embed parameter must not be promoted” assertion in `extract`.

Reply:
Dang, yeah. For whatever reason it seems that optional promotion does its thing before overload resolution?